src/core/core.scale.js

```diff
@@ -708,11 +713,28 @@ module.exports = Element.extend({
 
 		var itemsToDraw = [];
 
-		var axisWidth = helpers.valueAtIndexOrDefault(me.options.gridLines.lineWidth, 0);
-		var xTickStart = options.position === 'right' ? me.left : me.right - axisWidth - tl;
-		var xTickEnd = options.position === 'right' ? me.left + tl : me.right;
-		var yTickStart = options.position === 'bottom' ? me.top + axisWidth : me.bottom - tl - axisWidth;
-		var yTickEnd = options.position === 'bottom' ? me.top + axisWidth + tl : me.bottom + axisWidth;
+		var axisWidth = helpers.valueAtIndexOrDefault(gridLines.lineWidth, 0, 0);
+		var alignedLeft = helpers.alignLinePixel(chart, me.left, axisWidth);
+		var alignedRight = helpers.alignLinePixel(chart, me.right, axisWidth);
+		var alignedTop = helpers.alignLinePixel(chart, me.top, axisWidth);
+		var alignedBottom = helpers.alignLinePixel(chart, me.bottom, axisWidth);
+
+		var xTickStart, xTickEnd, yTickStart, yTickEnd;
```

Comment:
If I'm not wrong, depending on the scale orientation, only half of these variables (including the new `aligned*`) are effectively used so I think we could simplify for something like:

```javascript
var axisWidth = helpers.valueAtIndexOrDefault(gridLines.lineWidth, 0, 0);
var alignPixel = helpers._alignPixel;
var aligned, tickStart, tickEnd;

if (position === 'top') {
    aligned = alignPixel(chart, me.bottom, axisWidth);
    tickStart = me.bottom - tl;
    tickEnd = aligned - axisWidth / 2;
} else if (position === 'bottom') {
    aligned = alignPixel(chart, me.top, axisWidth);
    tickStart = aligned + axisWidth / 2;
    tickEnd = me.top + tl;
} else if (position === 'left') {
    aligned = alignPixel(chart, me.right, axisWidth);
    tickStart = me.right - tl;
    tickEnd = aligned - axisWidth / 2;
} else {
    aligned = alignPixel(chart, me.left, axisWidth);
    tickStart = aligned + axisWidth / 2;
    tickEnd = me.left + tl;
}
```

Feel free to rename `aligned` for whatever you think it's better.

Reply:
I like this simplification. I renamed it to `borderValue` as it stores the position of the axis border.